synapse/module_api/__init__.py

```diff
@@ -94,8 +96,16 @@ def invalidate_access_token(self, access_token):
         Raises:
             synapse.api.errors.AuthError: the access token is invalid
         """
-
-        return self._auth_handler.delete_access_token(access_token)
+        # see if the access token corresponds to a device
+        user_info = yield self._auth.get_user_by_access_token(access_token)
+        device_id = user_info.get("device_id")
+        user_id = user_info["user"].to_string()
+        if device_id:
+            # delete the device, which will also delete its access tokens
+            yield self.hs.get_device_handler().delete_device(user_id, device_id)
```

Comment:
So there's always going to be only one access token per device?

Reply:
yes, that's what we expect.